src/main/java/seedu/todo/ui/view/TaskCardView.java

```diff
@@ -121,12 +121,15 @@ private void displayTags(){
      */
     private void setStyle() {
         boolean isCompleted = task.isCompleted();
-        boolean isOverdue = task.getEndTime().isPresent() && timeUtil.isOverdue(task.getEndTime().get());
+        boolean isOverdue = task.getEndTime().isPresent() && timeUtil.isOverdue(task.getEndTime().get()) && !task.isEvent();
```

Comment:
The problem this is trying to solve is that overdue styling was also being applied to events, which is wrong - events can't be "overdue". Model side automatically marks "overdue" events as completed, but the thread is not synchronized to UI side's thread, so for a few seconds incorrect style would be applied to events. This avoids that problem